File: Python/fibonacci.py
# Find the Nth Fibonacci Number

class Solution:
    def fib(self, N: int) -> int:
        if N <= 1:
            return N
        previous, current = 0, 1
        for _ in range(N - 1):
            new_current = previous + current
            previous, current = current, new_current
        return current

File: C#/Fibonacci.cs
using System;
using System.Linq;
using System.Security.Principal;

class Fibonacci
{
    /// <summary>
    /// Compute nth value in Fibonacci sequence
    /// </summary>
    public static uint ComputeNth(uint i)
    {
        return (i < 2) ? i : ComputeNth(i, 2, 0, 1);
    }

    private static uint ComputeNth(uint i, int current, uint prev1, uint prev2)
    {
        var sum = prev1 + prev2;
        return (current >= i) ? sum : ComputeNth(i, current + 1, prev2, sum);
    }
}
